cassovary-core/src/test/scala/com/twitter/cassovary/util/io/MemoryMappedIntLongSourceSpec.scala

```diff
@@ -0,0 +1,49 @@
+package com.twitter.cassovary.util.io
+
+import java.io.{RandomAccessFile, DataOutputStream, FileOutputStream, File}
+
+import org.scalatest.{Matchers, WordSpec}
+
+/**
+ * Created by peter on 8/15/15.
+ */
+class MemoryMappedIntLongSourceSpec extends WordSpec with Matchers {
+  "MemoryMappedIntLongSource" should {
+    "read Ints and Longs from a multi-GB file" in {
+       // create a file with ints and long at various locations
+       val intValues = Map(
+         0L -> 12,
+         4L -> -34,
+         // Make sure ints on both sides of a buffer boundary are read correctly
+         (1L << 30) - 4 -> 56,
+         (1L << 30) -> 12345678
+       )
+      val longValues = Map(
+        (1L << 32) - 8 -> 98L,
+        (1L << 32) -> -76L,
+        (1L << 32) + 128 -> 123456789012345L
+      )
+      val file = File.createTempFile("IntLongData", "dat")
+      val out = new RandomAccessFile(file, "rw")
+      for ((i, value) <- intValues) {
+        out.seek(i)
+        out.writeInt(value)
+      }
+      for ((i, value) <- longValues) {
+        out.seek(i)
+        out.writeLong(value)
+      }
+      out.close()
+
```

Comment:
Good point.  I just added tests for nonexistent file and for index out of bounds.  The error messages for index out of bounds are currently not ideal (they refer to the index in the underlying buffer, not the invalid index the user gave us).  Do you think I should add explicit bounds checking to `getInt` and `getLong`?  This would make error messages more informative, but would add a bit of overhead to code that could be in an inner loop of any graph algorithm.